File: Patterns/code/java/DesignPatterns/src/main/java/visitor/Patron.java
package visitor;

class Patron implements Visitor {
    @Override
    public void visitPainting(Painting painting) {
        // Interact with the painting exhibit
        System.out.println("Patron admires the painting");
    }

    @Override
    public void visitSculpture(Sculpture sculpture) {
        // Interact with the sculpture exhibit
        System.out.println("Patron examines the sculpture");
    }
}

File: code/typescript/visitor/Patron.ts
namespace Visitor {
    export class Patron implements Visitor {
        visitPainting(painting: Painting): void {
            // Interact with the painting exhibit
            console.log("Patron admires the painting");
        }

        visitSculpture(sculpture: Sculpture): void {
            // Interact with the sculpture exhibit
            console.log("Patron examines the sculpture");
        }
    }
}
